// @aws-sdk/client-cognito-identity-provider/dist-cjs/CognitoIdentityProvider.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CognitoIdentityProvider = void 0;
const smithy_client_1 = require("@smithy/smithy-client");
const CognitoIdentityProviderClient_1 = require("./CognitoIdentityProviderClient");
const AddCustomAttributesCommand_1 = require("./commands/AddCustomAttributesCommand");
const AdminAddUserToGroupCommand_1 = require("./commands/AdminAddUserToGroupCommand");
const AdminConfirmSignUpCommand_1 = require("./commands/AdminConfirmSignUpCommand");
const AdminCreateUserCommand_1 = require("./commands/AdminCreateUserCommand");
const AdminDeleteUserAttributesCommand_1 = require("./commands/AdminDeleteUserAttributesCommand");
const AdminDeleteUserCommand_1 = require("./commands/AdminDeleteUserCommand");
const AdminDisableProviderForUserCommand_1 = require("./commands/AdminDisableProviderForUserCommand");
const AdminDisableUserCommand_1 = require("./commands/AdminDisableUserCommand");
const AdminEnableUserCommand_1 = require("./commands/AdminEnableUserCommand");
const AdminForgetDeviceCommand_1 = require("./commands/AdminForgetDeviceCommand");
const AdminGetDeviceCommand_1 = require("./commands/AdminGetDeviceCommand");
const AdminGetUserCommand_1 = require("./commands/AdminGetUserCommand");
const AdminInitiateAuthCommand_1 = require("./commands/AdminInitiateAuthCommand");
const AdminLinkProviderForUserCommand_1 = require("./commands/AdminLinkProviderForUserCommand");
const AdminListDevicesCommand_1 = require("./commands/AdminListDevicesCommand");
const AdminListGroupsForUserCommand_1 = require("./commands/AdminListGroupsForUserCommand");
const AdminListUserAuthEventsCommand_1 = require("./commands/AdminListUserAuthEventsCommand");
const AdminRemoveUserFromGroupCommand_1 = require("./commands/AdminRemoveUserFromGroupCommand");
const AdminResetUserPasswordCommand_1 = require("./commands/AdminResetUserPasswordCommand");
const AdminRespondToAuthChallengeCommand_1 = require("./commands/AdminRespondToAuthChallengeCommand");
const AdminSetUserMFAPreferenceCommand_1 = require("./commands/AdminSetUserMFAPreferenceCommand");
const AdminSetUserPasswordCommand_1 = require("./commands/AdminSetUserPasswordCommand");
const AdminSetUserSettingsCommand_1 = require("./commands/AdminSetUserSettingsCommand");
const AdminUpdateAuthEventFeedbackCommand_1 = require("./commands/AdminUpdateAuthEventFeedbackCommand");
const AdminUpdateDeviceStatusCommand_1 = require("./commands/AdminUpdateDeviceStatusCommand");
const AdminUpdateUserAttributesCommand_1 = require("./commands/AdminUpdateUserAttributesCommand");
const AdminUserGlobalSignOutCommand_1 = require("./commands/AdminUserGlobalSignOutCommand");
const AssociateSoftwareTokenCommand_1 = require("./commands/AssociateSoftwareTokenCommand");
const ChangePasswordCommand_1 = require("./commands/ChangePasswordCommand");
const ConfirmDeviceCommand_1 = require("./commands/ConfirmDeviceCommand");
const ConfirmForgotPasswordCommand_1 = require("./commands/ConfirmForgotPasswordCommand");
const ConfirmSignUpCommand_1 = require("./commands/ConfirmSignUpCommand");
const CreateGroupCommand_1 = require("./commands/CreateGroupCommand");
const CreateIdentityProviderCommand_1 = require("./commands/CreateIdentityProviderCommand");
const CreateResourceServerCommand_1 = require("./commands/CreateResourceServerCommand");
const CreateUserImportJobCommand_1 = require("./commands/CreateUserImportJobCommand");
const CreateUserPoolClientCommand_1 = require("./commands/CreateUserPoolClientCommand");
const CreateUserPoolCommand_1 = require("./commands/CreateUserPoolCommand");
const CreateUserPoolDomainCommand_1 = require("./commands/CreateUserPoolDomainCommand");
const DeleteGroupCommand_1 = require("./commands/DeleteGroupCommand");
const DeleteIdentityProviderCommand_1 = require("./commands/DeleteIdentityProviderCommand");
const DeleteResourceServerCommand_1 = require("./commands/DeleteResourceServerCommand");
const DeleteUserAttributesCommand_1 = require("./commands/DeleteUserAttributesCommand");
const DeleteUserCommand_1 = require("./commands/DeleteUserCommand");
const DeleteUserPoolClientCommand_1 = require("./commands/DeleteUserPoolClientCommand");
const DeleteUserPoolCommand_1 = require("./commands/DeleteUserPoolCommand");
const DeleteUserPoolDomainCommand_1 = require("./commands/DeleteUserPoolDomainCommand");
const DescribeIdentityProviderCommand_1 = require("./commands/DescribeIdentityProviderCommand");
const DescribeResourceServerCommand_1 = require("./commands/DescribeResourceServerCommand");
const DescribeRiskConfigurationCommand_1 = require("./commands/DescribeRiskConfigurationCommand");
const DescribeUserImportJobCommand_1 = require("./commands/DescribeUserImportJobCommand");
const DescribeUserPoolClientCommand_1 = require("./commands/DescribeUserPoolClientCommand");
const DescribeUserPoolCommand_1 = require("./commands/DescribeUserPoolCommand");
const DescribeUserPoolDomainCommand_1 = require("./commands/DescribeUserPoolDomainCommand");
const ForgetDeviceCommand_1 = require("./commands/ForgetDeviceCommand");
const ForgotPasswordCommand_1 = require("./commands/ForgotPasswordCommand");
const GetCSVHeaderCommand_1 = require("./commands/GetCSVHeaderCommand");
const GetDeviceCommand_1 = require("./commands/GetDeviceCommand");
const GetGroupCommand_1 = require("./commands/GetGroupCommand");
const GetIdentityProviderByIdentifierCommand_1 = require("./commands/GetIdentityProviderByIdentifierCommand");
const GetLogDeliveryConfigurationCommand_1 = require("./commands/GetLogDeliveryConfigurationCommand");
const GetSigningCertificateCommand_1 = require("./commands/GetSigningCertificateCommand");
const GetUICustomizationCommand_1 = require("./commands/GetUICustomizationCommand");
const GetUserAttributeVerificationCodeCommand_1 = require("./commands/GetUserAttributeVerificationCodeCommand");
const GetUserCommand_1 = require("./commands/GetUserCommand");
const GetUserPoolMfaConfigCommand_1 = require("./commands/GetUserPoolMfaConfigCommand");
const GlobalSignOutCommand_1 = require("./commands/GlobalSignOutCommand");
const InitiateAuthCommand_1 = require("./commands/InitiateAuthCommand");
const ListDevicesCommand_1 = require("./commands/ListDevicesCommand");
const ListGroupsCommand_1 = require("./commands/ListGroupsCommand");
const ListIdentityProvidersCommand_1 = require("./commands/ListIdentityProvidersCommand");
const ListResourceServersCommand_1 = require("./commands/ListResourceServersCommand");
const ListTagsForResourceCommand_1 = require("./commands/ListTagsForResourceCommand");
const ListUserImportJobsCommand_1 = require("./commands/ListUserImportJobsCommand");
const ListUserPoolClientsCommand_1 = require("./commands/ListUserPoolClientsCommand");
const ListUserPoolsCommand_1 = require("./commands/ListUserPoolsCommand");
const ListUsersCommand_1 = require("./commands/ListUsersCommand");
const ListUsersInGroupCommand_1 = require("./commands/ListUsersInGroupCommand");
const ResendConfirmationCodeCommand_1 = require("./commands/ResendConfirmationCodeCommand");
const RespondToAuthChallengeCommand_1 = require("./commands/RespondToAuthChallengeCommand");
const RevokeTokenCommand_1 = require("./commands/RevokeTokenCommand");
const SetLogDeliveryConfigurationCommand_1 = require("./commands/SetLogDeliveryConfigurationCommand");
const SetRiskConfigurationCommand_1 = require("./commands/SetRiskConfigurationCommand");
const SetUICustomizationCommand_1 = require("./commands/SetUICustomizationCommand");
const SetUserMFAPreferenceCommand_1 = require("./commands/SetUserMFAPreferenceCommand");
const SetUserPoolMfaConfigCommand_1 = require("./commands/SetUserPoolMfaConfigCommand");
const SetUserSettingsCommand_1 = require("./commands/SetUserSettingsCommand");
const SignUpCommand_1 = require("./commands/SignUpCommand");
const StartUserImportJobCommand_1 = require("./commands/StartUserImportJobCommand");
const StopUserImportJobCommand_1 = require("./commands/StopUserImportJobCommand");
const TagResourceCommand_1 = require("./commands/TagResourceCommand");
const UntagResourceCommand_1 = require("./commands/UntagResourceCommand");
const UpdateAuthEventFeedbackCommand_1 = require("./commands/UpdateAuthEventFeedbackCommand");
const UpdateDeviceStatusCommand_1 = require("./commands/UpdateDeviceStatusCommand");
const UpdateGroupCommand_1 = require("./commands/UpdateGroupCommand");
const UpdateIdentityProviderCommand_1 = require("./commands/UpdateIdentityProviderCommand");
const UpdateResourceServerCommand_1 = require("./commands/UpdateResourceServerCommand");
const UpdateUserAttributesCommand_1 = require("./commands/UpdateUserAttributesCommand");
const UpdateUserPoolClientCommand_1 = require("./commands/UpdateUserPoolClientCommand");
const UpdateUserPoolCommand_1 = require("./commands/UpdateUserPoolCommand");
const UpdateUserPoolDomainCommand_1 = require("./commands/UpdateUserPoolDomainCommand");
const VerifySoftwareTokenCommand_1 = require("./commands/VerifySoftwareTokenCommand");
const VerifyUserAttributeCommand_1 = require("./commands/VerifyUserAttributeCommand");
const commands = {
    AddCustomAttributesCommand: AddCustomAttributesCommand_1.AddCustomAttributesCommand,
    AdminAddUserToGroupCommand: AdminAddUserToGroupCommand_1.AdminAddUserToGroupCommand,
    AdminConfirmSignUpCommand: AdminConfirmSignUpCommand_1.AdminConfirmSignUpCommand,
    AdminCreateUserCommand: AdminCreateUserCommand_1.AdminCreateUserCommand,
    AdminDeleteUserCommand: AdminDeleteUserCommand_1.AdminDeleteUserCommand,
    AdminDeleteUserAttributesCommand: AdminDeleteUserAttributesCommand_1.AdminDeleteUserAttributesCommand,
    AdminDisableProviderForUserCommand: AdminDisableProviderForUserCommand_1.AdminDisableProviderForUserCommand,
    AdminDisableUserCommand: AdminDisableUserCommand_1.AdminDisableUserCommand,
    AdminEnableUserCommand: AdminEnableUserCommand_1.AdminEnableUserCommand,
    AdminForgetDeviceCommand: AdminForgetDeviceCommand_1.AdminForgetDeviceCommand,
    AdminGetDeviceCommand: AdminGetDeviceCommand_1.AdminGetDeviceCommand,
    AdminGetUserCommand: AdminGetUserCommand_1.AdminGetUserCommand,
    AdminInitiateAuthCommand: AdminInitiateAuthCommand_1.AdminInitiateAuthCommand,
    AdminLinkProviderForUserCommand: AdminLinkProviderForUserCommand_1.AdminLinkProviderForUserCommand,
    AdminListDevicesCommand: AdminListDevicesCommand_1.AdminListDevicesCommand,
    AdminListGroupsForUserCommand: AdminListGroupsForUserCommand_1.AdminListGroupsForUserCommand,
    AdminListUserAuthEventsCommand: AdminListUserAuthEventsCommand_1.AdminListUserAuthEventsCommand,
    AdminRemoveUserFromGroupCommand: AdminRemoveUserFromGroupCommand_1.AdminRemoveUserFromGroupCommand,
    AdminResetUserPasswordCommand: AdminResetUserPasswordCommand_1.AdminResetUserPasswordCommand,
    AdminRespondToAuthChallengeCommand: AdminRespondToAuthChallengeCommand_1.AdminRespondToAuthChallengeCommand,
    AdminSetUserMFAPreferenceCommand: AdminSetUserMFAPreferenceCommand_1.AdminSetUserMFAPreferenceCommand,
    AdminSetUserPasswordCommand: AdminSetUserPasswordCommand_1.AdminSetUserPasswordCommand,
    AdminSetUserSettingsCommand: AdminSetUserSettingsCommand_1.AdminSetUserSettingsCommand,
    AdminUpdateAuthEventFeedbackCommand: AdminUpdateAuthEventFeedbackCommand_1.AdminUpdateAuthEventFeedbackCommand,
    AdminUpdateDeviceStatusCommand: AdminUpdateDeviceStatusCommand_1.AdminUpdateDeviceStatusCommand,
    AdminUpdateUserAttributesCommand: AdminUpdateUserAttributesCommand_1.AdminUpdateUserAttributesCommand,
    AdminUserGlobalSignOutCommand: AdminUserGlobalSignOutCommand_1.AdminUserGlobalSignOutCommand,
    AssociateSoftwareTokenCommand: AssociateSoftwareTokenCommand_1.AssociateSoftwareTokenCommand,
    ChangePasswordCommand: ChangePasswordCommand_1.ChangePasswordCommand,
    ConfirmDeviceCommand: ConfirmDeviceCommand_1.ConfirmDeviceCommand,
    ConfirmForgotPasswordCommand: ConfirmForgotPasswordCommand_1.ConfirmForgotPasswordCommand,
    ConfirmSignUpCommand: ConfirmSignUpCommand_1.ConfirmSignUpCommand,
    CreateGroupCommand: CreateGroupCommand_1.CreateGroupCommand,
    CreateIdentityProviderCommand: CreateIdentityProviderCommand_1.CreateIdentityProviderCommand,
    CreateResourceServerCommand: CreateResourceServerCommand_1.CreateResourceServerCommand,
    CreateUserImportJobCommand: CreateUserImportJobCommand_1.CreateUserImportJobCommand,
    CreateUserPoolCommand: CreateUserPoolCommand_1.CreateUserPoolCommand,
    CreateUserPoolClientCommand: CreateUserPoolClientCommand_1.CreateUserPoolClientCommand,
    CreateUserPoolDomainCommand: CreateUserPoolDomainCommand_1.CreateUserPoolDomainCommand,
    DeleteGroupCommand: DeleteGroupCommand_1.DeleteGroupCommand,
    DeleteIdentityProviderCommand: DeleteIdentityProviderCommand_1.DeleteIdentityProviderCommand,
    DeleteResourceServerCommand: DeleteResourceServerCommand_1.DeleteResourceServerCommand,
    DeleteUserCommand: DeleteUserCommand_1.DeleteUserCommand,
    DeleteUserAttributesCommand: DeleteUserAttributesCommand_1.DeleteUserAttributesCommand,
    DeleteUserPoolCommand: DeleteUserPoolCommand_1.DeleteUserPoolCommand,
    DeleteUserPoolClientCommand: DeleteUserPoolClientCommand_1.DeleteUserPoolClientCommand,
    DeleteUserPoolDomainCommand: DeleteUserPoolDomainCommand_1.DeleteUserPoolDomainCommand,
    DescribeIdentityProviderCommand: DescribeIdentityProviderCommand_1.DescribeIdentityProviderCommand,
    DescribeResourceServerCommand: DescribeResourceServerCommand_1.DescribeResourceServerCommand,
    DescribeRiskConfigurationCommand: DescribeRiskConfigurationCommand_1.DescribeRiskConfigurationCommand,
    DescribeUserImportJobCommand: DescribeUserImportJobCommand_1.DescribeUserImportJobCommand,
    DescribeUserPoolCommand: DescribeUserPoolCommand_1.DescribeUserPoolCommand,
    DescribeUserPoolClientCommand: DescribeUserPoolClientCommand_1.DescribeUserPoolClientCommand,
    DescribeUserPoolDomainCommand: DescribeUserPoolDomainCommand_1.DescribeUserPoolDomainCommand,
    ForgetDeviceCommand: ForgetDeviceCommand_1.ForgetDeviceCommand,
    ForgotPasswordCommand: ForgotPasswordCommand_1.ForgotPasswordCommand,
    GetCSVHeaderCommand: GetCSVHeaderCommand_1.GetCSVHeaderCommand,
    GetDeviceCommand: GetDeviceCommand_1.GetDeviceCommand,
    GetGroupCommand: GetGroupCommand_1.GetGroupCommand,
    GetIdentityProviderByIdentifierCommand: GetIdentityProviderByIdentifierCommand_1.GetIdentityProviderByIdentifierCommand,
    GetLogDeliveryConfigurationCommand: GetLogDeliveryConfigurationCommand_1.GetLogDeliveryConfigurationCommand,
    GetSigningCertificateCommand: GetSigningCertificateCommand_1.GetSigningCertificateCommand,
    GetUICustomizationCommand: GetUICustomizationCommand_1.GetUICustomizationCommand,
    GetUserCommand: GetUserCommand_1.GetUserCommand,
    GetUserAttributeVerificationCodeCommand: GetUserAttributeVerificationCodeCommand_1.GetUserAttributeVerificationCodeCommand,
    GetUserPoolMfaConfigCommand: GetUserPoolMfaConfigCommand_1.GetUserPoolMfaConfigCommand,
    GlobalSignOutCommand: GlobalSignOutCommand_1.GlobalSignOutCommand,
    InitiateAuthCommand: InitiateAuthCommand_1.InitiateAuthCommand,
    ListDevicesCommand: ListDevicesCommand_1.ListDevicesCommand,
    ListGroupsCommand: ListGroupsCommand_1.ListGroupsCommand,
    ListIdentityProvidersCommand: ListIdentityProvidersCommand_1.ListIdentityProvidersCommand,
    ListResourceServersCommand: ListResourceServersCommand_1.ListResourceServersCommand,
    ListTagsForResourceCommand: ListTagsForResourceCommand_1.ListTagsForResourceCommand,
    ListUserImportJobsCommand: ListUserImportJobsCommand_1.ListUserImportJobsCommand,
    ListUserPoolClientsCommand: ListUserPoolClientsCommand_1.ListUserPoolClientsCommand,
    ListUserPoolsCommand: ListUserPoolsCommand_1.ListUserPoolsCommand,
    ListUsersCommand: ListUsersCommand_1.ListUsersCommand,
    ListUsersInGroupCommand: ListUsersInGroupCommand_1.ListUsersInGroupCommand,
    ResendConfirmationCodeCommand: ResendConfirmationCodeCommand_1.ResendConfirmationCodeCommand,
    RespondToAuthChallengeCommand: RespondToAuthChallengeCommand_1.RespondToAuthChallengeCommand,
    RevokeTokenCommand: RevokeTokenCommand_1.RevokeTokenCommand,
    SetLogDeliveryConfigurationCommand: SetLogDeliveryConfigurationCommand_1.SetLogDeliveryConfigurationCommand,
    SetRiskConfigurationCommand: SetRiskConfigurationCommand_1.SetRiskConfigurationCommand,
    SetUICustomizationCommand: SetUICustomizationCommand_1.SetUICustomizationCommand,
    SetUserMFAPreferenceCommand: SetUserMFAPreferenceCommand_1.SetUserMFAPreferenceCommand,
    SetUserPoolMfaConfigCommand: SetUserPoolMfaConfigCommand_1.SetUserPoolMfaConfigCommand,
    SetUserSettingsCommand: SetUserSettingsCommand_1.SetUserSettingsCommand,
    SignUpCommand: SignUpCommand_1.SignUpCommand,
    StartUserImportJobCommand: StartUserImportJobCommand_1.StartUserImportJobCommand,
    StopUserImportJobCommand: StopUserImportJobCommand_1.StopUserImportJobCommand,
    TagResourceCommand: TagResourceCommand_1.TagResourceCommand,
    UntagResourceCommand: UntagResourceCommand_1.UntagResourceCommand,
    UpdateAuthEventFeedbackCommand: UpdateAuthEventFeedbackCommand_1.UpdateAuthEventFeedbackCommand,
    UpdateDeviceStatusCommand: UpdateDeviceStatusCommand_1.UpdateDeviceStatusCommand,
    UpdateGroupCommand: UpdateGroupCommand_1.UpdateGroupCommand,
    UpdateIdentityProviderCommand: UpdateIdentityProviderCommand_1.UpdateIdentityProviderCommand,
    UpdateResourceServerCommand: UpdateResourceServerCommand_1.UpdateResourceServerCommand,
    UpdateUserAttributesCommand: UpdateUserAttributesCommand_1.UpdateUserAttributesCommand,
    UpdateUserPoolCommand: UpdateUserPoolCommand_1.UpdateUserPoolCommand,
    UpdateUserPoolClientCommand: UpdateUserPoolClientCommand_1.UpdateUserPoolClientCommand,
    UpdateUserPoolDomainCommand: UpdateUserPoolDomainCommand_1.UpdateUserPoolDomainCommand,
    VerifySoftwareTokenCommand: VerifySoftwareTokenCommand_1.VerifySoftwareTokenCommand,
    VerifyUserAttributeCommand: VerifyUserAttributeCommand_1.VerifyUserAttributeCommand,
};
class CognitoIdentityProvider extends CognitoIdentityProviderClient_1.CognitoIdentityProviderClient {
}
exports.CognitoIdentityProvider = CognitoIdentityProvider;
(0, smithy_client_1.createAggregatedClient)(commands, CognitoIdentityProvider);
